data/transactions/logic/resources.go

```diff
@@ -153,6 +153,9 @@ func (r *resources) fillAssetTransfer(hdr *transactions.Header, tx *transactions
 
 // allowsHolding checks if a holding is available under the txgroup sharing rules
 func (cx *EvalContext) allowsHolding(addr basics.Address, ai basics.AssetIndex) bool {
+	if cx.UnlimitedResourceAccess {
+		return true
+	}
```

Comment:
I did my best to put checks on `cx.UnlimitedResourceAccess` as late as possible so that they wouldn't be executed unless the resource really isn't available. I was able to do that everywhere except for the two occurrences in this file.

The problem here is that we immediately return the output of `cx.availableAsset/App`. These will be false for unavailable assets even if `cx.UnlimitedResourceAccess` is true (`resolveApp/Asset` and `assignApp/Asset` are where the changes for unlimited resources happen, to preserve slot indexing behavior).

If we really wanted to push the `cx.UnlimitedResourceAccess` to the bottom of this function, we could if we changed our statements from the form `return cx.availableAsset(ai)` to

```go
if cx.availableAsset(ai) {
    return true
}
```

I thought that was a bit odd though, so I didn't do it, but happy to hear other opinions.

Reply:
I think the change to putting the unlimit check in `availableXXX()` would address this, right?

Reply:
Correct, resolving